### PR TITLE
Update nodeport-proxy to use envoyv3

### PR DESCRIPTION
**Pull request by youssefazrak**
**What this PR does / why we need it**:
As part of the short-term strategy for the new expose strategy, we will use the existing Nodeport-proxy.
This one needs to be updated to use v3 packages.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by youssefazrak** [comment]
/override pre-kubermatic-spellcheck
See https://github.com/vmware-tanzu/velero/pull/3063

File: go.mod [hunk 1]
```diff
@@ -99,7 +99,8 @@ replace (
 	github.com/go-logr/logr => github.com/go-logr/logr v0.2.1-0.20200730175230-ee2de8da5be6
 	github.com/go-logr/zapr => github.com/go-logr/zapr v0.2.0
 	// pin to prevent panics in DeepEqual() when comparing envoy objects
```
Comment (irozzo-1A) [inline go.mod]: I would remove the entire comment as we are now using `reflect.DeepEqual` and this no longer applies.

Comment (youssefazrak) [reply go.mod]: Good point, it's fixed now.

**Pull request merged**